flink-table-store-connector/src/test/java/org/apache/flink/table/store/connector/FileStoreITCase.java

```diff
@@ -0,0 +1,286 @@
+
+import javax.annotation.Nullable;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.apache.flink.table.store.file.FileStoreOptions.BUCKET;
+import static org.apache.flink.table.store.file.FileStoreOptions.FILE_FORMAT;
+import static org.apache.flink.table.store.file.FileStoreOptions.FILE_PATH;
+import static org.assertj.core.api.Assertions.assertThat;
+
+/** ITCase for {@link FileStoreSource} and {@link StoreSink}. */
+@RunWith(Parameterized.class)
+public class FileStoreITCase extends AbstractTestBase {
+
+    private static final RowType PARTITION_TYPE =
+            new RowType(Collections.singletonList(new RowType.RowField("p", new VarCharType())));
+
+    private static final RowType KEY_TYPE =
+            new RowType(Collections.singletonList(new RowType.RowField("k", new IntType())));
+
+    private static final RowType VALUE_TYPE =
+            new RowType(
+                    Arrays.asList(
+                            new RowType.RowField("v", new IntType()),
+                            new RowType.RowField("p", new VarCharType()),
+                            // rename key
+                            new RowType.RowField("_k", new IntType())));
+
+    @SuppressWarnings({"unchecked", "rawtypes"})
+    public static final DataStructureConverter<RowData, Row> CONVERTER =
+            (DataStructureConverter)
+                    DataStructureConverters.getConverter(
+                            TypeConversions.fromLogicalToDataType(VALUE_TYPE));
+
+    private static final int NUM_BUCKET = 3;
+
+    private static final List<RowData> SOURCE_DATA =
+            Arrays.asList(
+                    wrap(GenericRowData.of(0, StringData.fromString("p1"), 1)),
+                    wrap(GenericRowData.of(0, StringData.fromString("p1"), 2)),
+                    wrap(GenericRowData.of(5, StringData.fromString("p1"), 1)),
+                    wrap(GenericRowData.of(6, StringData.fromString("p2"), 1)),
+                    wrap(GenericRowData.of(3, StringData.fromString("p2"), 5)),
+                    wrap(GenericRowData.of(5, StringData.fromString("p2"), 1)));
+
+    private final boolean isBatch;
+
+    public FileStoreITCase(boolean isBatch) {
+        this.isBatch = isBatch;
+    }
+
+    @Parameterized.Parameters(name = "isBounded-{0}")
+    public static List<Boolean> getVarSeg() {
+        return Arrays.asList(true, false);
+    }
+
+    private static SerializableRowData wrap(RowData row) {
+        return new SerializableRowData(row, InternalSerializers.create(VALUE_TYPE));
+    }
+
+    @Test
+    public void testPartitioned() throws Exception {
+        innerTest(true);
+    }
+
+    @Test
+    public void testNonPartitioned() throws Exception {
+        innerTest(false);
+    }
+
+    @Test
+    public void testOverwrite() throws Exception {
+        if (!isBatch) {
```

Comment:
Nit: you can try `Assume.assumeTrue(isBatch)` to conditionally skip the test
Ref: https://junit.org/junit4/javadoc/4.13/org/junit/Assume.html